src/commonMain/kotlin/teksturepako/pakku/api/projects/Project.kt

```diff
@@ -182,6 +182,13 @@ data class Project(
         return this.files.filter { it.type in providers.map { provider -> provider.serialName } }
     }
 
+    fun getFilesForProviders(providers: Collection<Provider>): List<ProjectFile>
+    {
+        return this.files.filter { it.type in providers.map { provider -> provider.serialName } }
+    }
+
+    fun getLatestFile(providers: Collection<Provider>) = getFilesForProviders(providers).maxByOrNull { it.datePublished }
```

Comment:
Only a question, what would happen when the datePublished is the default value?

Reply:
natural order